### PR TITLE
atari800_libretro: an Atari 8bit computer + 5200 console emulator

### PR DESCRIPTION
**Pull request by kwyxz**
This is a brand new recipe to build atari800_libretro, a libretro core providing an Atari 8bit emulator

This emulator covers
- Atari 400/800/1200 + XL computers
- the 5200 console

### CODE REVIEW

File: app-emulation/atari800_libretro/atari800_libretro-3.1.0.20171023git.recipe [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+emulator, originally written by David Firth and now developed by the Atari800 \
+Development Team."
+HOMEPAGE="https://atari800.github.io/"
+COPYRIGHT="2001-2018, the Atari800 Dev team, the libretro team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+srcGitRev="f892b4437d52a3e1c4b42b18f3edc564a1d1045b"
+SOURCE_URI="https://github.com/libretro/libretro-atari800/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="0c412610237e10b518373b5fd0f9bfe743ae03b288122f899887ea745905949a"
+SOURCE_FILENAME="libretro-atari800-$portVersion-$srcGitRev.tar.gz"
```
Comment (fbrosson) [inline app-emulation/atari800_libretro/atari800_libretro-3.1.0.20171023git.recipe]:
```diff
-SOURCE_FILENAME="libretro-atari800-$portVersion-$srcGitRev.tar.gz"
+SOURCE_FILENAME="libretro-atari800-${portVersion/_/-}-$srcGitRev.tar.gz"
```

File: app-emulation/atari800_libretro/atari800_libretro-3.1.0.20171023git.recipe [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	sed -e "s/@DISPLAY_VERSION@/v${portVersion%git}/" \
```
Comment (fbrosson) [inline app-emulation/atari800_libretro/atari800_libretro-3.1.0.20171023git.recipe]:
```diff
-	sed -e "s/@DISPLAY_VERSION@/v${portVersion%git}/" \
+	sed -e "s/@DISPLAY_VERSION@/v${portVersion/_/-}-${srcGitRev:0:7}/" \
```
(Do we insert a `-${srcGitRev:0:7}` here? Or is it not necessary?)

Comment (kwyxz) [reply app-emulation/atari800_libretro/atari800_libretro-3.1.0.20171023git.recipe]: Not necessary, the one inserted after make is a simple fix to help avoiding a build error without having to use PATCH()

File: app-emulation/atari800_libretro/atari800_libretro-3.1.0_20171023.recipe [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+emulator, originally written by David Firth and now developed by the Atari800 \
+Development Team."
+HOMEPAGE="https://atari800.github.io/"
+COPYRIGHT="2001-2018, the Atari800 Dev team, the libretro team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+srcGitRev="f892b4437d52a3e1c4b42b18f3edc564a1d1045b"
+SOURCE_URI="https://github.com/libretro/libretro-atari800/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="0c412610237e10b518373b5fd0f9bfe743ae03b288122f899887ea745905949a"
+SOURCE_FILENAME="libretro-atari800-$portVersion-$srcGitRev.tar.gz"
```
Comment (fbrosson) [inline app-emulation/atari800_libretro/atari800_libretro-3.1.0_20171023.recipe]: Suggestion:
```diff
-SOURCE_FILENAME="libretro-atari800-$portVersion-$srcGitRev.tar.gz"
+SOURCE_FILENAME="libretro-atari800-${portVersion/_/-}-$srcGitRev.tar.gz"
```

File: app-emulation/atari800_libretro/atari800_libretro-3.1.0_20171023.recipe [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	zlib${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	lib:libz$secondaryArchSuffix
```
Comment (fbrosson) [inline app-emulation/atari800_libretro/atari800_libretro-3.1.0_20171023.recipe]:
```diff
 BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
-	zlib${secondaryArchSuffix}_devel
+	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
 	cmd:gcc$secondaryArchSuffix
 	cmd:make
-	lib:libz$secondaryArchSuffix
 	"
```
BTW, I guess you will need to insert a `lib:libz$secondaryArchSuffix` in `REQUIRES`.

You can run:
```
objdump -x /system/add-ons/libretro/atari800_libretro.so | grep NEEDED
```
to see if it needs libz.
Maybe haikuporter's `-S` switch only check this for files in `$binDir` and `$libDir`.

**Comment by kwyxz** [comment]
Fixed zlib dependencies in latest commit

File: app-emulation/atari800_libretro/atari800_libretro-3.1.0_20171023.recipe [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	retroarch$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libz${secondaryArchSuffix}
```
Comment (fbrosson) [inline app-emulation/atari800_libretro/atari800_libretro-3.1.0_20171023.recipe]:
```diff
-	devel:libz${secondaryArchSuffix}
+	devel:libz$secondaryArchSuffix
```
Sorry for this request ;)
Thanks!

**Comment by kwyxz** [comment]
Whoops, my bad. Fixed.

**Comment by fbrosson** [comment]
I just noticed that the recipe does not include a `PATCHES` variable.
But the build was OK.
In case you add the patchset, don't forget to also rename it from `atari800_libretro-3.1.0.20171023.patchset` to `atari800_libretro-3.1.0_20171023.patchset` (if you add `PATCHES="atari800_libretro-$portVersion.patchset`) or to `atari800_libretro-3.1.0-20171023.patchset` (if you add `PATCHES="atari800_libretro-${portVersion/_/-}.patchset"`)

**Comment by kwyxz** [comment]
That's funny. Does haikuporter apply the patchset even without the PATCHES variable?
I will add it nonetheless, but it should not have built without it.

**Comment by fbrosson** [comment]
No, it does not apply the patchset, even if the name matches the recipe.
It only applies the patchset if the recipe has a `PATCHES`.

That said, if you write a recipe, make changes to the sources, and build, it will build with your changes.
This means that if you export your changes with `haikuporter -e atari800_libretro` then it's a good idea to clean the work directory and kick a fresh build. This way, if you the build fails during a fresh build you will know that this is (probably) because you forgot to mention the patchset in a `PATCHES` variable.

There is one thing haikuporter spots about the patches: if a patchset exists with a name that perfectly matches the recipe name, and if you forgot to mention the patchset in the recipe, then haikuporter will display a warning about it.

**Comment by kwyxz** [comment]
Well it looks like it builds without the patchset now. That's interesting. I guess I'll remove it then!

**Comment by kwyxz** [comment]
Not sure why it didn't build with my initial attempt but it works fine now.

**Pull request merged**